test/riak_kv_sweeper_SUITE.erl

```diff
@@ -207,16 +309,62 @@ test_scheduler_worker_process_crashed(Config) ->
     ok = check_sweeps_idle(Indices),
     ok.
 
+%%
+%% Test riak_kv_sweeper's re-scheduling of sweeps, configured by the
+%% run_interval property. run_interval is an integer.
+%%
+%% The test uses sweep participant sweeper_callback_1 that completes
+%% successfully as soon as it is scheduled.
+%%
+%% The test asserts receipt of successful_sweep messages after sending
+%% sweep_ticks and then waiting run_interval+1 seconds, and then
+%% re-sending sweep_tick's.
+%%
 test_scheduler_run_interval(Config) ->
     Indices = ?config(vnode_indices, Config),
-    new_sweep_participant(sweeper_callback_1, 1),
+    RunInterval = 1,
+    new_sweep_participant(sweeper_callback_1, RunInterval),
+    sweeper_meck_helper:backend_modules(async, _NumKeys=1000, _ObjecSizeBytes=0),
+
+    sweep_all_indices(Indices),
+    timer:sleep(timer:seconds(RunInterval + 1)),
+    sweep_all_indices(Indices),
+    ok.
+
+%%
+%% Test riak_kv_sweeper's re-scheduling of sweeps, configured by the
+%% run_interval property. run_interval is a fun.
+%%
+%% The test uses sweep participant sweeper_callback_1 that completes
+%% successfully as soon as it is scheduled.
+%%
+%% The test asserts receipt of successful_sweep messages after sending
+%% sweep_ticks and then waiting run_interval+1 seconds, and then
+%% re-sending sweep_tick's.
+%%
+test_scheduler_run_interval_as_fun(Config) ->
+    Indices = ?config(vnode_indices, Config),
+    RunIntervalFun = fun() -> 1 end,
+    new_sweep_participant(sweeper_callback_1, RunIntervalFun),
     sweeper_meck_helper:backend_modules(async, _NumKeys=1000, _ObjecSizeBytes=0),
 
     sweep_all_indices(Indices),
-    timer:sleep(min_scheduler_response_time_msecs()),
+    timer:sleep(timer:seconds(RunIntervalFun() + 1)),
```

Comment:
Something that just occurred to me: we have a couple tests now that verify that sweeps run after the run interval has expired, but do we have any tests that verify the inverse? We may also want to test that sweeps _don't_ run again until the run interval has elapsed. Perhaps if we could rework the tests to get rid of the sleep, then maybe that's a test we could add at the same time (e.g. if we use something like wait_until, we could track how long it had to wait, and make sure it had to wait at least as long as the run_interval).